rust/template/distributed_datalog/src/sources/file.rs

```diff
@@ -176,15 +176,15 @@ where
 {
     /// Create a new adapter streaming data from the file at the given
     /// `path`.
-    pub fn new<P>(path: P) -> Result<Self, Error>
+    pub fn new<P>(path: P) -> Self
```

Comment:
Such a check is currently happening in `subscribe`. In my opinion it makes more sense there, because the corresponding `unsubscribe` closes the file. A constructor is invoked only once for an object, but at least conceptually we support multiple subscriptions, so the file's life cycle gets awkward if we create/open it in the constructor, I believe.